### PR TITLE
added book controller, dto for book and createRequest, added MapStruct

### PR DESCRIPTION
**Pull request by holsoni**

### CODE REVIEW

File: src/main/java/com/example/bookstore/config/MapperConfig.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
+        componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl"
+)
+public class MapperConfig {
```
Comment (CodeWebWeaver) [inline src/main/java/com/example/bookstore/config/MapperConfig.java]:
```suggestion
@MapperConfig(
        componentModel = "spring",
        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
        implementationPackage = "<PACKAGE_NAME>.impl"
)
public class MapperConfiguration {
```

**Review by CodeWebWeaver — CHANGES_REQUESTED** [review]
Nice solution let`s make one change!

**Review by CodeWebWeaver — APPROVED** [review]
Nice

File: src/main/java/com/example/bookstore/config/MapperConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import org.mapstruct.NullValueCheckStrategy;
+
+@MapperConfig(
+        componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl"
+)
+public class MapperConfiguration {
+
```
Comment (nacenik) [inline src/main/java/com/example/bookstore/config/MapperConfiguration.java]: Empty line

File: src/main/java/com/example/bookstore/mapper/BookMapper.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import com.example.bookstore.dto.BookDto;
+import com.example.bookstore.dto.CreateBookRequestDto;
+import com.example.bookstore.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfiguration.class)
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    Book toModel(CreateBookRequestDto requestDto);
```
Comment (nacenik) [inline src/main/java/com/example/bookstore/mapper/BookMapper.java:15]: I'd recommend you add `@Mapping(target = "id", ignore = true)` above this method to show, that the `id` has to be ignored

File: src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -42,11 +43,21 @@ public Book save(Book book) {
     }
 
     @Override
-    public List<Book> findAll() {
+    public List<Book> getAll() {
         try (Session session = sessionFactory.openSession()) {
             return session.createQuery("FROM Book", Book.class).getResultList();
         } catch (Exception e) {
-            throw new RuntimeException("Can't get all books from the DB", e);
+            throw new RuntimeException("Can't get all books from the DB!", e);
```
Comment (nacenik) [inline src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java]: better do not throw common `RuntimeException` and create custom exception

File: src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -42,11 +43,21 @@ public Book save(Book book) {
...
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.find(Book.class, id));
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get book with id {"
```
Comment (nacenik) [inline src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java]: the same

File: src/main/java/com/example/bookstore/service/BookService.java [hunk 6]
```diff
@@ -1,10 +1,13 @@
 package com.example.bookstore.service;
 
-import com.example.bookstore.model.Book;
+import com.example.bookstore.dto.BookDto;
+import com.example.bookstore.dto.CreateBookRequestDto;
 import java.util.List;
 
 public interface BookService {
-    Book save(Book book);
+    BookDto createBook(CreateBookRequestDto createBookRequestDto);
```
Comment (nacenik) [inline src/main/java/com/example/bookstore/service/BookService.java]:
```suggestion
    BookDto create(CreateBookRequestDto createBookRequestDto);
```

**Review by nacenik — CHANGES_REQUESTED** [review]
Good, let's improve your solution)

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**